hack/lib/util/golang.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+#
+# This library holds golang related utility functions.
+
+# Ensure the go tool exists and is a viable version.
+os::golang::verify_go_version() {
```

Comment:
To explain: the formatting is for consistency, the readonly is for scripts that call other scripts where both `source hack/lib/init.sh`